### ISSUE TITLE
ANgular build require Path on build

### ISSUE THREAD

**Issue by Mehdi-MST** [swagger-api/swagger-ui#9856]
when we build Angular app it gives an error on  node_modules/swagger-ui-dist/absolute-path.js:9:19:

const getAbsoluteFSPath = function () {
  // detect whether we are running in a browser or nodejs
  if (typeof module !== "undefined" && module.exports) {
    return require("path").resolve(__dirname)
  }
  throw new Error('getAbsoluteFSPath can only be called within a Nodejs environment');
}

export default getAbsoluteFSPath

by removing the line return require("path").resolve(__dirname) it solve the issue . any other possible fix for that please?

**Comment by Mehdi-MST** [comment]
I am using the  lastest version 5.16.2 and also the version before that had the  same problem

**Comment by Mehdi-MST** [comment]
using try catch solved the issue
const getAbsoluteFSPath = function () {
  // detect whether we are running in a browser or nodejs
  if (typeof module !== "undefined" && module.exports) {
    try {
      return resolve(__dirname)
    }
    catch (e) {
      throw new Error('Unable to determine the current working directory')
    }
  }
}

module.exports = getAbsoluteFSPath

**Comment by samerkassem82** [comment]
I have the same issue, anyone?

**Comment by Mehdi-MST** [comment]
Yes I use angular. Doesnt matter ssr or not on build I get that error. Temporary fix it comment that line and everything works fine

**Comment by samerkassem82** [comment]
@Mehdi-MST  For me it wasn't resolved , when I commented it

```
/*
 * getAbsoluteFSPath
 * @return {string} When run in NodeJS env, returns the absolute path to the current directory
 *                  When run outside of NodeJS, will return an error message
 */
const getAbsoluteFSPath = function () {
  // detect whether we are running in a browser or nodejs
  if (typeof module !== "undefined" && module.exports) {
   // return require("path").resolve(__dirname)
  }
  throw new Error('getAbsoluteFSPath can only be called within a Nodejs environment');
}

module.exports = getAbsoluteFSPath
```


I got Error: getAbsoluteFSPath can only be called within a Nodejs environment

**Comment by Mehdi-MST** [comment]
When you comment out that line and run build that should solve the issue. But whenever you run npm i that overwrite that line again. So that doesnt help if you are using automation builder

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by BeataKr** [comment]
I'm facing the same issue - angular v17, @Mehdi-MST did you manage to resolve problem in any different way, than commenting-out the line?

**Comment by Aloxion** [comment]
Facing same issue, any updates on this?
